torchgeo/datamodules/landcoverai.py

```diff
@@ -36,6 +38,43 @@ def __init__(
         self.batch_size = batch_size
         self.num_workers = num_workers
 
+    def on_after_batch_transfer(
+        self, batch: Dict[str, Any], batch_idx: int
+    ) -> Dict[str, Any]:
+        """Apply batch augmentations after batch is transferred to the device.
+
+        Args:
+            batch: mini-batch of data
+            batch_idx: batch index
+
+        Returns:
+            augmented mini-batch
+        """
+        try:
+            if self.trainer.training:  # type: ignore[union-attr]
+                # Kornia expects masks to be floats with a channel dimension
+                x = batch["image"]
+                y = batch["mask"].float().unsqueeze(1)
+
+                train_augmentations = K.AugmentationSequential(
+                    K.RandomRotation(p=0.5, degrees=90),
+                    K.RandomHorizontalFlip(p=0.5),
+                    K.RandomVerticalFlip(p=0.5),
+                    K.RandomSharpness(p=0.5),
+                    K.ColorJitter(
+                        p=0.5, brightness=0.1, contrast=0.1, saturation=0.1, hue=0.1
+                    ),
+                    data_keys=["input", "mask"],
+                )
+                x, y = train_augmentations(x, y)
+
+                # torchmetrics expects masks to be longs without a channel dimension
+                batch["image"] = x
+                batch["mask"] = y.squeeze(1).long()
+        except AttributeError:
+            pass
```

Comment:
What is this except for? This seems dangerous

Reply:
This is needed when the datamodule is used without a trainer. We could instead replace this with:
```python
if hasattr(self, "trainer") and hasattr(self.trainer, "training"):
```
Let me know if you want me to change this.

Reply:
yes that is less dangerous I think?

Reply:
Done